components/layout_2020/flow/construct.rs

```diff
@@ -158,25 +162,46 @@ impl BlockContainer {
         }
 
         type Intermediate<Node> = IntermediateBlockLevelBox<Node>;
-        #[derive(Default)]
         struct Target {
             contains_floats: ContainsFloats,
+            outer_content_sizes_of_children: ContentSizes,
+        }
+        impl Default for Target {
+            fn default() -> Self {
+                Self {
+                    contains_floats: ContainsFloats::No,
+                    outer_content_sizes_of_children: ContentSizes::zero(),
+                }
+            }
         }
         let mut target = Target {
             contains_floats: builder.contains_floats,
+            outer_content_sizes_of_children: ContentSizes::zero(),
         };
-        let request_content_sizes = false; // FIXME
+        let request_childrens_outer_content_sizes = request_content_sizes;
         let iter = builder.block_level_boxes.into_par_iter();
         let iter = iter.mapfold_reduce_into(
             &mut target,
             |target, (intermediate, box_slot): (Intermediate<_>, BoxSlot<'_>)| {
-                let (block_level_box, box_contains_floats) =
-                    intermediate.finish(context, request_content_sizes);
+                let (block_level_box, box_contains_floats) = intermediate.finish(
+                    context,
+                    if request_childrens_outer_content_sizes {
+                        Some(&mut target.outer_content_sizes_of_children)
+                    } else {
+                        None
+                    },
+                );
                 target.contains_floats |= box_contains_floats;
                 box_slot.set(LayoutBox::BlockLevel(block_level_box.clone()));
                 block_level_box
             },
-            |left, right| left.contains_floats |= right.contains_floats,
+            |left, right| {
+                left.contains_floats |= right.contains_floats;
+                if request_content_sizes {
+                    left.outer_content_sizes_of_children
+                        .max_assign(&right.outer_content_sizes_of_children)
+                }
```

Comment:
I do not think this is compatible with how we handle margin collapsing.

Reply:
This only computes sizes in the inline direction, margin-collapsing only affects the block direction.